src/libs/estimate/reestimate.ts

```diff
@@ -0,0 +1,31 @@
+export async function reestimate(fetchRequests: Function, counter: number = 0): Promise<any> {
+  // stop the execution on 5 fails;
+  // the below error message is not shown to the user so we are safe
+  if (counter >= 5)
+    return new Error(
+      'Estimation failure, retrying in a couple of seconds. If this issue persists, please change your RPC provider or contact Ambire support'
+    )
+
+  const estimationTimeout = new Promise((resolve) => {
+    setTimeout(() => {
+      resolve('Timeout reached')
```

Comment:
you can do this more elegantly without having to check for typeof

the way it's done is
```
// allocate an object that will be resulted from the promise that can then be compared by reference
const santinelTimeoutErr = {}

...
resolve(santinelTimeoutErr)
....

if (result === santinelTimeoutErr)

```